### DISCUSSION TITLE
How to asynchronously deserialize data when using createWebStoragePersister

### DISCUSSION THREAD

**Discussion by aubreyzulu** [TanStack/query#3692] [Q&A]
` import { QueryClient } from 'react-query';
 import { persistQueryClient } from 'react-query/persistQueryClient'
 import { createWebStoragePersister } from 'react-query/createWebStoragePersister'
 
 import { compress, decompress } from 'lz-string';
 
 const queryClient = new QueryClient({ defaultOptions: { queries: { staleTime: Infinity } } });
 
 persistQueryClient({
   queryClient: connectionsQueryClient,
   persistor: createWebStoragePersister({
     storage: window.localStorage,
     serialize: data => compress(JSON.stringify(data)),
     deserialize: data => JSON.parse(decompress(data)),
   }),
   maxAge: Infinity,
 });`

**Comment by TkDodo** [comment]
I don't think its possible at the moment. however, `lz-string` [seems to be synchronous](https://github.com/pieroxy/lz-string/blob/2f749bc9fe3cc889fa9e32d2769a930d977596e6/typings/lz-string.d.ts#L13-L14) anyways?
